lib/bundler/installer.rb

```diff
@@ -119,6 +119,8 @@ def generate_bundler_executable_stubs(spec, options = {})
       relative_gemfile_path = relative_gemfile_path
       ruby_command = Thor::Util.ruby_command
       ruby_command = ruby_command
+      application_locked = options[:locked] || Bundler.settings[:application_locked]
```

Comment:
`Bundler.feature_flag.application_locked?`

Reply:
Amended the commit with this change.

Reply:
Actually, the use of `Bundler.settings[:application_locked]` is just returning whether the `:application_locked` setting for this run of the installer as opposed to returning the state of the feature. For example, if using Bundler >= 2 and using `--locked=false` with the `binstub` command.